### PR TITLE
Switch workflows to target `main` only

### PR DESCRIPTION
**Pull request by cqc-alec**
# Description

Update CI workflows to run on push and PR to `main` instead of `develop`.

Once approved, I would:
- change the branch protection rules on the repo (note that some of the "Required" checks below will not run on this PR);
- merge this into `develop`;
- make `main` the default branch;
- make a PR from `develop` into `main`;
- merge that.

We will also need to update the workflows on `tket-site`, so that they only publish released versions (which will be identifiable by tag).

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/docs/backends.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ Contains :py:class:`Backend` abstract class and associated methods. In pytket a
 
 There are several `example notebooks <https://tket.quantinuum.com/examples>`_ on pytket :py:class:`Backend`\s. If you are interested in developing your own :py:class:`Backend` or pytket extension then see the `creating backends <https://tket.quantinuum.com/examples/creating_backends.html>`_ tutorial.
 
-Notebook tutorials specific to the :py:class:`QuantinuumBackend` can be found `here <https://github.com/CQCL/pytket-quantinuum/tree/develop/examples>`_.
+Notebook tutorials specific to the :py:class:`QuantinuumBackend` can be found `here <https://github.com/CQCL/pytket-quantinuum/tree/main/examples>`_.
```
Comment (cqc-melf) [inline pytket/docs/backends.rst:8]: Is this change intended? We have not updated that repo yet?

**Review by cqc-melf — COMMENTED** [review]
can you maybe outline what the plan for the releases is?
Do we want to create a release from main or from a release branch?

**Comment by cqc-melf** [comment]
Have you checked if there are any open PRs that might be effected by this?

File: pytket/docs/backends.rst (hunk 1 above)
Comment (cqc-alec) [reply pytket/docs/backends.rst:8]: You're right, that was a mistake, I'll revert it.

**Comment by cqc-alec** [comment]
> can you maybe outline what the plan for the releases is? Do we want to create a release from main or from a release branch?

We can do either, but normally we will just release from main, as at present. If we want to do a hotfix release that doesn't have all the commits from main then we will create a special release branch for that.

**Comment by cqc-alec** [comment]
> Have you checked if there are any open PRs that might be effected by this?

This should just be a matter of changing the target branch in the PR.

**Review by cqc-melf — APPROVED** [review]
